src/Fixer/ClassNotation/ConstTypeRequiredFixer.php

```diff
@@ -0,0 +1,385 @@
+
+            return;
+        }
+
+        // all other `const` cases were we cannot reliably and safely determine the type of
+        $this->insertTypeForConst($tokens, $constIndex + 1, 'mixed');
+    }
+
+    private function fixSimpleConstant(
+        Tokens $tokens,
+        int $constIndex,
+        int $valueIndex
+    ): void {
+        if ($tokens[$valueIndex]->isGivenKind(T_STRING)) {
+            $this->fixSimpleStringConstant($tokens, $constIndex, $valueIndex);
+
+            return;
+        }
+
+        $token = $tokens[$valueIndex];
+        $type = null;
+
+        if ($token->isGivenKind(T_LNUMBER)) {
+            $type = 'int';
+        } elseif ($token->isGivenKind(T_DNUMBER)) {
+            $type = 'float';
+        } elseif ($token->isGivenKind(T_CONSTANT_ENCAPSED_STRING)) {
+            $type = 'string';
+        } elseif ($this->isMagicConstOfTypeStr($token)) {
+            $type = 'string';
+        } elseif ($this->isMagicConstOfTypeInt($token)) {
+            $type = 'int';
+        }
+
+        $this->insertTypeForConst($tokens, $constIndex + 1, $type ?? 'mixed');
+    }
+
+    private function fixSimpleStringConstant(
+        Tokens $tokens,
+        int $constIndex,
+        int $valueIndex
+    ): void {
+        $type = $this->getTypeForStringToken($tokens[$valueIndex]);
+        $this->insertTypeForConst($tokens, $constIndex + 1, $type ?? 'mixed');
+    }
+
+    private function fixArrayConstant(
+        Tokens $tokens,
+        int $constIndex,
+        int $valueIndex
+    ): void {
+        if ($tokens[$valueIndex]->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_OPEN)) {
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_ARRAY_SQUARE_BRACE, $valueIndex);
+        } else {
+            \assert($tokens[$valueIndex]->isGivenKind(T_ARRAY));
+            $endIndex = $tokens->getNextMeaningfulToken($valueIndex);
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $endIndex);
+        }
+
+        $endIndex = $tokens->getNextMeaningfulToken($endIndex);
+        $this->insertTypeForConst($tokens, $constIndex + 1, $tokens[$endIndex]->equals(';') ? 'array' : 'mixed');
+    }
+
+    private function insertTypeForConst(Tokens $tokens, int $index, string $type): void
+    {
+        $tokens->insertSlices(
+            [
+                $index => [
+                    new Token([T_WHITESPACE, ' ']),
+                    new Token(['array' === $type ? CT::T_ARRAY_TYPEHINT : T_STRING, $type]),
+                ],
+            ],
+        );
+    }
+
+    private function getTypeForStringToken(Token $token): ?string
+    {
+        \assert($token->isGivenKind(T_STRING));
+
+        $content = $token->getContent();
+        $contentLower = strtolower($content);
+
+        if ('true' === $contentLower || 'false' === $contentLower) {
+            $type = 'bool';
+        } elseif ('null' === $contentLower) {
+            $type = 'null';
```

Comment:
```suggestion
        if ('true' === $contentLower || 'false' === $contentLower || 'null' === $contentLower) {
            $type = $contentLower;
```

Reply:
thanks, done in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7352/commits/cd68b4192a344189951816ecbcba1ccf802018f2